go/auth/token.go

```diff
@@ -0,0 +1,163 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+//
+// Code used to support authentication tokens for arbitrary purposes.
+//
+package auth
+
+import (
+	"bytes"
+	"encoding/json"
+	"math"
+	"time"
+
+	libkb "github.com/keybase/client/go/libkb"
+	keybase1 "github.com/keybase/client/go/protocol"
+)
+
+const CurrentTokenVersion = 1
+
+type TokenKey struct {
+	UID      keybase1.UID             `json:"uid"`
+	Username libkb.NormalizedUsername `json:"username"`
+	KID      keybase1.KID             `json:"kid"`
+}
+
+type TokenBody struct {
+	Key     TokenKey `json:"key"`
+	Type    string   `json:"type"`
+	Version int      `json:"version"`
+}
+
+type TokenClient struct {
+	Name    string `json:"name"`
+	Version string `json:"version"`
+}
+
+type Token struct {
+	Body         TokenBody   `json:"body"`
+	Client       TokenClient `json:"client"`
+	CreationTime int64       `json:"ctime"`
+	ExpireIn     int         `json:"expire_in"`
+	Tag          string      `json:"tag"`
+}
```

Comment:
See the comment in the PR (I put it there so we don't lose it after follow-on commits). This auth token still doesn't match the one we're currently using for the client<->API server auth.  Can you fix it so it does?